src/planning/strategies/match-recipe-by-verb.ts

```diff
@@ -29,12 +31,15 @@ import {StrategizerWalker, Strategy} from '../strategizer.js';
 // Note that the recipe may have the slot pattern multiple times over, but
 // this strategy currently only connects the first instance of the pattern up
 // if there are multiple instances.
+type HandleConstraint = {handle: Handle, direction: Direction};
+type SlotConstraint = {targetSlot: Slot, providedSlots: Dict<Slot>};
+
 export class MatchRecipeByVerb extends Strategy {
 
-  async generate(inputParams) {
+  async generate(inputParams: GenerateParams<Recipe>): Promise<Descendant<Recipe>[]> {
     const arc = this.arc;
     return StrategizerWalker.over(this.getResults(inputParams), new class extends StrategizerWalker {
-      onParticle(recipe: Recipe, particle: Particle) {
+      onParticle(_recipe: Recipe, particle: Particle) {
```

Comment:
Why rename to `_recipe` here and in other places?

Reply:
This tells tslint that the value is intentionally not used, otherwise there's a warning with no-unused-variables on. https://github.com/palantir/tslint/issues/1337 suggests that this may no longer be a part of tslint in v4.